### PR TITLE
Fixes getting free quirk points through prosthetic limb quirks

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

~~Makes prothetic limbs granted through Prothetic Limb and Quadruple Amputee quirks unremovable through surgery(also can't be removed via amputation shears) and adds a trait for making bodyparts unremovable by surgeries~~

Makes prosthetic limbs granted through Prothetic Limb and Quadruple Amputee quirks give you downsides if replaced by non-surplus limbs. Replaced legs will make you limp, and replaced hands will make you have a small chance to drop held items.

## Why It's Good For The Game

Right now you can just take one of this quirks, and then run roundstart to a roboticist to get it replaced by a normal robotic limb. Most other negative quirks downsides are pernament, so there is no reason to give a person 3/8 quirk points just for waiting 1 minute while a surgeon is replacing their surplus limbs with good ones. With this PR you wound't be able to do this no more.

## Changelog

:cl:
balance: Prothetic Limb and Quadruple Amputee quirks now will cripple you if limbs granted by them are replaced with non-surplus ones
/:cl:

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
I forsee a slight bump in self-welderbombings or fireaxe tidings in the near future.

**Comment by timothymtorres** [comment]
I'd like to point out that someone can still just take a knife and cutoff the limb on hostile intent.

**Comment by SuperSlayer0** [comment]
> I'd like to point out that someone can still just take a knife and cutoff the limb on hostile intent.

Will fix that

**Comment by TemporalOroboros** [comment]
Does it count as a negative quirk at this point? Now it means you can't get dismembered. At least not for that limb.

**Comment by SuperSlayer0** [comment]
> Does it count as a negative quirk at this point? Now it means you can't get dismembered. At least not for that limb.

It can still be disabled by damage, and has only 20 health, so yeah, it is still a negative quirk. Through I still can remove the dismemberment part of this PR or lower the quirk points grained by this quirks(if needed).

**Comment by timothymtorres** [comment]
> > I'd like to point out that someone can still just take a knife and cutoff the limb on hostile intent.
> 
> Will fix that

I was pointing it out not so you can fix it, but to show that your changes can by bypassed quite easily.  Also by making the limbs unable to be cutoff, it's actually somewhat of a blessing as now in combat all of your limbs have immunity to being removed.  This was one of the reasons prosthetics were at a disadvantage to normal limbs since they were weaker.

I'm not sure what the correct solution to the problem would be honestly.

**Comment by Jacquerel** [comment]
I know quirks aren’t usually reversible but when we need to go so far as to make the “shitty prosthetic” also be attached to you at an unbreakable fundamental level you really have to ask whether it matters that much.

**Comment by SuperSlayer0** [comment]
> it's actually somewhat of a blessing as now in combat all of your limbs have immunity to being removed. 

Okay, maybe then I will just keep it only unremovable through surgery. At least it will fix one of exploits rather then all of them.

**Comment by Mothblocks** [comment]
this sounds like its going to be very very easy to regress.

i think it would be a lot simpler for us if we made it so getting your prosthetic limb replaced with something else crippled you in some way. like we can make you walk with a limp if your prosthetic gets replaced and say its because youre not used to it, or make your replaced arm drop stuff

**Comment by SuperSlayer0** [comment]
> this sounds like its going to be very very easy to regress.
> 
> i think it would be a lot simpler for us if we made it so getting your prosthetic limb replaced with something else crippled you in some way. like we can make you walk with a limp if your prosthetic gets replaced and say its because youre not used to it, or make your replaced arm drop stuff

It sounds better then my current solution, I will check if I am able to do so.

**Comment by Mothblocks** [comment]
:+1: we are the community where the meta used to be to suicide with a box to be borged, we definitely would never be able to whack a mole this before

**Comment by zxaber** [comment]
Seems somewhat unnecessary if the replacement limb is a borg limb; you still has the largest drawback of the perk, being the something like 15 second stun from EMPs. It's also kinda gamey in general that the limb is somehow impervious to surgery tools.

The bigger exploit is probably replacing the limb with an organic one (possibly donated from the spare roundstart morgue bodies).

**Comment by SuperSlayer0** [comment]
> Seems somewhat unnecessary if the replacement limb is a borg limb; you still have the largest drawback of the perk, being the something like 15 second stun from EMPs. 

Can't agree, normal robotic limbs have damage reduction and have much more health then surplus ones.

File: code/datums/quirks/negative_quirks.dm [hunk 1]
```diff
@@ -528,21 +531,55 @@
...
+	for(var/obj/item/bodypart/leg/leg in human_holder.bodyparts)
+		if(checked_limb_type && !istype(bodypart, checked_limb_type))
+			continue
+		if(is_type_in_typecache(bodypart, limb_typecahce))
+			continue
+		if(prob(50))
+			limping_message = TRUE
+			human_holder.client.move_delay += limp_slowdown
+	if(limping_message && prob(20)) ///Don't want to spam this
+		to_chat(human_holder, "You limp on one of your legs, due to not getting used to them.")
```
Comment (Mothblocks) [inline code/datums/quirks/negative_quirks.dm:579]: we already have a nice limp in wounds, any way we can reuse the code for that?

**Comment by Wallemations** [comment]
I always thought the downside of prosthetic limbs was having to carry around wire and a welder with you? Wouldn't having cyborg limbs also count towards that?

**Comment by SuperSlayer0** [comment]
> I always thought the downside of prosthetic limbs was having to carry around wire and a welder with you? Wouldn't having cyborg limbs also count towards that?

It will, but the main downside of surplus limbs that are gained through this quirk is that fact that they have like 20 health, while cyborg limbs and organic ones have a lot more.

**Comment by RaveRadbury** [comment]
I don't see why we should punish cooperation and job content.

If you have a problem with the quirk being too easy, make it give fewer points.

**Comment by SuperSlayer0** [comment]
> I don't see why we should punish cooperation and job content.
> 
> If you have a problem with the quirk being too easy, make it give fewer points. 

Job cooperation is cool, but negative quirks should have permanent debuffs, since in exchange of them you get permanent buffs(mood buffs that speed you up, night vision, etc.). Otherwise you just get points for laying down on a operating table for few minutes (very fun gameplay) and then forget about that you had a negative quirk for the rest of the round.

**Comment by zxaber** [comment]
I would much rather this than nerfing the points given. It already got nerfed once.

**Comment by RaveRadbury** [comment]
> > I don't see why we should punish cooperation and job content.
> > If you have a problem with the quirk being too easy, make it give fewer points.
> 
> Job cooperation is cool, but negative quirks should have permanent debuffs, since in exchange of them you get permanent buffs(mood buffs that speed you up, night vision, etc.). Otherwise you just get points for laying down on a operating table for few minutes (very fun gameplay) and then forget about that you had a negative quirk for the rest of the round.

I don't see getting a replacement limb as a guaranteed thing.

There should be some sort of advantage to engaging in cooperation and also providing potential job content/conflict with roboticists. I guess I can get behind the idea of "breaking in a new limb" or whatever, it's not like the sturdiness of the upgraded parts is changing.

**Comment by newfren** [comment]
this feels like it's fixing a problem that doesn't exist.

**Comment by SuperSlayer0** [comment]
> this feels like it's fixing a problem that doesn't exist. 

I literally was myself a person who was asked by a lizard with this quirk to get their arm replaced with a normal one.

**Comment by newfren** [comment]
I do not see that as a problem

**Comment by SuperSlayer0** [comment]
> I do not see that as a problem

So you think that getting 8 quirk points for which you can get, for example, night vision, in exchange of just waiting on a surgery table for one-two minutes isn't a problem?

**Comment by Draggeru** [comment]
>Right now you can just take one of this quirks, and then run roundstart to a roboticist to get it replaced by a normal robotic limb.

I have legitimately never seen people do this round start, even then why not just make it a 0 point quirk if it's actually a problem?

**Comment by SuperSlayer0** [comment]
> I have legitimately never seen people do this round start, even then why not just make it a 0 point quirk if it's actually a problem?

With 0 points you will just get some limbs that die in two hits, you get stunned by EMPs in exchange of... nothing.

**Comment by newfren** [comment]
> > I do not see that as a problem
> 
> So you think that getting 8 quirk points for which you can get, for example, night vision, in exchange of just waiting on a surgery table for one-two minutes isn't a problem?

no, I don't.

**Comment by ATH1909** [comment]
> > I do not see that as a problem
> 
> So you think that getting 8 quirk points for which you can get, for example, night vision, in exchange of just waiting on a surgery table for one-two minutes isn't a problem?

oh no, they have the night vision quirk now, whatever shall we doooooo?

there's very little to actually spend your positive quirk points on (that's worthwhile) currently, and I have my doubts that that will ever change.

**Comment by necromanceranne** [comment]
> > I do not see that as a problem
> 
> So you think that getting 8 quirk points for which you can get, for example, night vision, in exchange of just waiting on a surgery table for one-two minutes isn't a problem?

As a robo main, I welcome people coming to get their limbs replaced. Seems fine by me, and I get to have a chat with someone.

**Comment by SuperSlayer0** [comment]
Okay, if so many people don't like this...

**Pull request closed without merging**